com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/BindingUtils.java

```diff
@@ -33,46 +35,21 @@ private BindingUtils() {
      */
     public static String getMethodName(IMethodBinding binding, boolean fromKey) {
         if (fromKey) {
-            String key = binding.getKey();
-            int dotAt = key.indexOf('.');
-            int end = key.indexOf('<', dotAt);
-            if (end == -1) {
-                end = key.indexOf('(');
-            } else {
-                end = Math.min(end, key.indexOf('('));
-            }
-            return key.substring(dotAt + 1, end);
+            return LambdaLocationLocatorHelper.toMethodName(binding);
```

Comment:
Is there any case that the helper covers better?

Reply:
Yes it handles the method resolutions better without depending on the binding key if I remember correctly. 

I know its a internal class, but i guess rather duplicating the same logic we could look at making it public if we want to avoid internal api usage in future. 

WDYT ?

Reply:
Actually the implementation here is a little different with the helper method.

`Git blame` shows that it's this PR https://github.com/microsoft/java-debug/commit/69230b339d68dfa2b3a1c251c29f3bd577fd0cf8#diff-91a75ad1996749ac4383bad13f73aaab20a962d82966486c06a09bb6dd86d5ff that changed the implementation a little bit. Do you remember the history why you changed it last time?

Reply:
Its for handling keys which contains generics. basically strip the generics parts from the method name.